backend/helpers/analyticsRouteHandlerHelpers.js

```diff
@@ -362,6 +514,45 @@ const getGroupAnalyticsEffort = async (unitCode, year, period, groupNumber) => {
     return effortAnalyticsData;
 };
 
+const getGroupAnalyticsPreference = async (unitCode, year, period, groupNumber) => {
+    /**
+     * Given a group, obtains all information related to student preference
+     * to group matching
+     */
+
+    const preferenceAnalyticsData = {
+        "personality title": "Preference Selection",
+        "description":
+            "Based on the input of each student, our algorithm attempts to a" +
+            "ssign each student their most preferred group. This is an aggre" +
+            "gation of how many students were assigned to their nth preferre" +
+            "d selection.",
+        data: [],
+    };
+
+    /** const preferenceResults = await promiseBasedQuery("FILL THIS SQL CODE IN THANKS" [unitCode, year, period, groupNumber]) */
+
+    const preferenceBarChartData = {
+        type: "bar",
+        title: "Number of students assigned to their nth preferred group",
+        "x label": "Student effort",
+        "y label": "Number of students",
+        x: [],
+        y: [],
+    };
+
+    // if (preferenceResults.length > 0) {
+    //     preferenceResults.forEach((result) => {
+    //          preferenceBarChartData["x"].push(result["preference"]);
+    //          preferenceBarChartData["y"].push(result["count"]);
+    //     });
+
+    //     preferenceAnalyticsData["data"].push(preferenceBarChartData);
+    // }
+    
+    return preferenceAnalyticsData
```

Comment:
Why are we returning preferenceBarChartData without filling out the x and y arrays? This function doesn't look complete?

Reply:
This function was pulled from the analytics branch and should be replaced by the complete one that they have once they  do a pr as well

Reply:
Problem handled by analytics branch.